### PR TITLE
kubeadm: remove dead code

### PR DESCRIPTION
**Pull request by bart0sh**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remove unused functions and variables from the kubeadm codebase.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks @bart0sh 
/priority backlog

File: cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go [hunk 1]
```diff
@@ -29,14 +29,6 @@ import (
...
 	markcontrolplanephase "k8s.io/kubernetes/cmd/kubeadm/app/phases/markcontrolplane"
 	uploadconfigphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/uploadconfig"
-	"k8s.io/kubernetes/pkg/util/normalizer"
-)
-
-var (
-	controlPlaneJoinExample = normalizer.Examples(`
-		# Joins a machine as a control plane instance
-		kubeadm join phase control-plane-join all
-		`)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go]: i think we should add this as an example for the command instead:
`Example: ....`

File: cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go [hunk 2]
```diff
@@ -163,27 +163,3 @@ func runControlPlanePrepareKubeconfigPhaseLocal(c workflow.RunData) error {
...
-
-		cfg, err := data.InitCfg()
-		if err != nil {
-			return err
-		}
-
-		// Creates static pod manifests file for the control plane components to be deployed on this node
-		// Static pods will be created and managed by the kubelet as soon as it starts
-		fmt.Printf("[control-plane-prepare] Creating static Pod manifest for %q\n", component)
-		return controlplane.CreateStaticPodFiles(kubeadmconstants.GetStaticPodDirectory(), &cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, component)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: by removing this function, in this same file please rename:
`runControlPlaneSubphase` to `runControlPlanePrepareManifestsSubphase`

**Review by neolit123 — COMMENTED** [review]
@kubernetes/sig-cluster-lifecycle-pr-reviews

File: cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go (hunk 2 above)
Comment (bart0sh) [reply cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go]: done

**Comment by bart0sh** [comment]
@neolit123 I've changed the code as you've suggested. Can you review/lgtm/approve ?

File: cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go [hunk 3]
```diff
@@ -63,9 +55,10 @@ func getControlPlaneJoinPhaseFlags() []string {
 // NewControlPlaneJoinPhase creates a kubeadm workflow phase that implements joining a machine as a control plane instance
 func NewControlPlaneJoinPhase() workflow.Phase {
 	return workflow.Phase{
-		Name:  "control-plane-join",
-		Short: "Joins a machine as a control plane instance",
-		Long:  cmdutil.MacroCommandLongDescription,
+		Name:    "control-plane-join",
+		Short:   "Joins a machine as a control plane instance",
+		Long:    cmdutil.MacroCommandLongDescription,
+		Example: "kubeadm join phase control-plane-join all",
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go]: please look at similar phase files.
we should keep the `controlPlaneJoinExample` variable for consistency and just add it here:
`Example: controlPlaneJoinExample`

how usefully this is, is another question...but for consistency.

**Comment by neolit123** [comment]
> I've changed the code as you've suggested. Can you review/lgtm/approve ?

will wait on someone else to have a look too.

**Review by yagonobre — APPROVED** [review]
Thanks @bart0sh 
after addressing @neolit123 comments, LGTM

**Comment by bart0sh** [comment]
@yagonobre done

**Review by neolit123 — APPROVED** [review]
/approve

**Pull request merged**